File: snippets/fsharp/System/Object/ReferenceEquals/referenceequals4.fs
module referenceequals4

open System

// <Snippet1>
let int1 = 3
printfn $"{Object.ReferenceEquals(int1, int1)}"
printfn $"{int1.GetType().IsValueType}"

// The example displays the following output:
//       False
//       True
// </Snippet1>

File: snippets/csharp/System/Object/ReferenceEquals/referenceequals4.cs
using System;

public class ObjectReferenceEqualsExample1
{
    public static void Run()
    {
        // <Snippet1>
        int int1 = 3;
        Console.WriteLine(object.ReferenceEquals(int1, int1));
        Console.WriteLine(int1.GetType().IsValueType);

        // The example displays the following output:
        //       False
        //       True
        // </Snippet1>
    }
}
